samples/detect.js

```diff
@@ -52,43 +68,43 @@ function detectTextIntent(projectId, sessionId, queries, languageCode) {
       },
     };
 
-    if (!promise) {
-      // First query.
-      console.log(`Sending query "${query}"`);
-      promise = sessionClient.detectIntent(request);
-    } else {
-      promise = promise.then(responses => {
-        console.log('Detected intent');
-        const response = responses[0];
-        logQueryResult(sessionClient, response.queryResult);
-
-        // Use output contexts as input contexts for the next query.
-        response.queryResult.outputContexts.forEach(context => {
-          // There is a bug in gRPC that the returned google.protobuf.Struct
-          // value contains fields with value of null, which causes error
-          // when encoding it back. Converting to JSON and back to proto
-          // removes those values.
-          context.parameters = struct.encode(struct.decode(context.parameters));
-        });
-        request.queryParams = {
-          contexts: response.queryResult.outputContexts,
-        };
-
-        console.log(`Sending query "${query}"`);
-        return sessionClient.detectIntent(request);
-      });
+    // If we have contexts from previous interactions with the bot add them here
+    if (contexts && contexts.length > 0) {
+      request.queryParams = {
+        contexts: contexts,
+      };
     }
-  }
 
-  promise
-    .then(responses => {
-      console.log('Detected intent');
-      logQueryResult(sessionClient, responses[0].queryResult);
-    })
-    .catch(err => {
-      console.error('ERROR:', err);
-    });
+    const responses = await sessionClient.detectIntent(request);
+    return responses[0];
+  }
 
+  async function executeQueries(projectId, sessionId, queries, languageCode) {
+    // Keeping the context across queries let's us simulate an ongoing conversation with the bot
+    let context;
+    let intentResponse;
+    for (const query in queries) {
```

Comment:
`queries` is an array, we should use `for...of` instead of `for...in` statement. 

Rule of thumb: don't use `for...in`. Use `for...of` for arrays and `Object.entries()` or `Object.keys()` for object. `for...in` includes properties of the prototype object and has random order.

Reply:
Thanks for pointing that out.